raiden/tests/integration/long_running/test_token_networks.py

```diff
@@ -101,12 +101,12 @@ def test_participant_selection(raiden_network, token_addresses):
             joinable_funds_target=-1,
         )
 
-    # connect the other nodes
+    # Make all nodes connect
     connect_greenlets = [
         gevent.spawn(
             RaidenAPI(app.raiden).token_network_connect, registry_address, token_address, 100
         )
-        for app in raiden_network[1:]
```

Comment:
Since node 0 is already connecting, and only created 1 channel to (0x222...), why doesn't it automatically connect to the new nodes without calling `RaidenAPI(app.raiden).token_network_connect` again?

Reply:
Check the description of the commit: https://github.com/raiden-network/raiden/commit/d31ee5c4b4e243a87891e935a8a020008c812da0

I explained it there. Essentially looking at the code: https://github.com/raiden-network/raiden/blob/140c3e7b1b99844c6537f03717233fecff3ae695/raiden/connection_manager.py#L169-L180

You can see that the first connect simply bootstraps the network and does nothing else since there is no other node to connect to at the time.

But ... after looking at the code and running the test again without this change but with my bugfix it passed since `retry_connect` is called whenever a new channel is seen on the network and a `connection_manager` exists. Will amend the test in a new commit.